### ISSUE TITLE
Upgrading from 24.1.6 to 24.1.7 fails via Homebrew

### ISSUE THREAD

**Issue by rockchalkwushock** [erlang/otp#5439]
**Describe the bug**
Hello I really had no idea where to report this so I decided to drop it here on the GitHub repository as I don't think it has to do with `homebrew`. I just went to upgrade `erlang` to 24.1.7 and I get the following output. I've never encountered a problem with `erlang` before 🤷🏻 :

I am running:

```text
brew: 3.3.5
macOS: 12.0.1 (Intel Mac)
```

```shell
==> Upgrading 1 outdated package:
erlang 24.1.6 -> 24.1.7
==> Downloading https://ghcr.io/v2/homebrew/core/erlang/manifests/24.1.7
Already downloaded: /Users/codybrunner/Library/Caches/Homebrew/downloads/21d45df19a60030f6d420cbbd0b5e96026f5a4f65c3165ba8fbd5d89aa6639a8--erlang-24.1.7.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/erlang/blobs/sha256:d5adeb2b9d41f74c4ba51b2e12ccabdc6d94ccb4690efd8edae
Already downloaded: /Users/codybrunner/Library/Caches/Homebrew/downloads/421cbdc77f30341ab686bd78185412407b0c8a96df199542e95519e84092f55a--erlang--24.1.7.monterey.bottle.tar.gz
==> Upgrading erlang
  24.1.6 -> 24.1.7

==> Pouring erlang--24.1.7.monterey.bottle.tar.gz
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/unicode_usage.html: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/unicode_usage.html'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/uri_string.html: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/uri_string.html'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/uri_string_usage.html: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/uri_string_usage.html'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/users_guide.html: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/users_guide.html'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/win32reg.html: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/win32reg.html'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/zip.html: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/html/zip.html'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/pdf/: Can't create 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/pdf'
erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/pdf/stdlib-3.16.1.pdf: Failed to create dir 'erlang/24.1.7/lib/erlang/lib/stdlib-3.16.1/doc/pdf'
.....
.....
.....
The output was insanely long, basically all internals of Erlang fail to install.
....
tar: Error exit delayed from previous errors.
```

**To Reproduce**

```shell
brew upgrade
```

**Expected behavior**
Install and cleanup with no issues.

**Affected versions**
Seems like this is related to the tar/binary of 24.1.7

**Additional context**

When I ran `brew doctor` I received the following say that `Erlang` was unlinked:

```shell
Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  erlang
```

**Comment by rickard-green** [comment]
My guess is that this is an homebrew issue and not an Erlang/OTP issue, but it is hard to say from this limited information. These printouts do not give much information about what could have gone wrong. If you can provide more detailed build logs we can perhaps help say more about what went wrong. I'm not a user of homebrew and do not know any details of homebrew so I can unfortunately not help you with where to find such build logs.

**Comment by rickard-green** [comment]
More guessing. I guess that you are just fetching tar-balls that already should have been built and that your upgrade only unpacks these already built tar-balls onto your machine. That is, nothing is actually built on your machine. If this is the case, then this is either a homebrew issue with a broken build, or an issue with your installation (e.g. missing write access to directories written to). The reason why the build is broken could potentially be due to an Erlang/OTP issue (although I would be surprised if such a build issue was introduced in a patch). Anyway, someone with access to build logs needs to provide those to us if we are going to be able to help with any build issues.

**Comment by rockchalkwushock** [comment]
@rickard-green I apologize for the delayed response and want to thank you for the speedy response to the issue. I will provide the full logs here. I uninstalled `elixir` and `erlang` (installed as dependency to `elixir`). I re-installed `elixir` and it pulled in the latest version 24.1.7 of `erlang` and the same logs ensued....

I just tried a re-install of `elixir` again and it worked 😕 

Those damn gremlins in the inter tubes are at it again!

Thank you again for your quick response, closing.
